tools/reviewer/web/src/app/dashboard/reviews-panel/review/review.component.html

```diff
@@ -88,59 +236,172 @@
       </div>
     </div>
 
+    <!-- Issue -->
+    <div 
+    fxLayout="row"
+    fxLayoutAlign="start center"
+    fxLayout.xs="column"
+    fxLayoutGap.xs="10px"
+    class="bug">
+    <div fxFlex="150px" fxFlex.xs="" 
+      fxLayoutAlign="start center" 
+      class="property">
+      Issue:
+    </div>
+    <div fxFlex class="detail">
+      <ng-container>
+        <div
+          class="issue-detail"
+          *ngIf="!showEditableBug"
+          fxLayoutAlign="start center">
+          <a class="link" href="{{bug}}">
+            {{ bug }}
+          </a>
+          <button 
+            mat-icon-button
+            (click)="showEditableBug = true"
+            *ngIf="notEditable"
+            >
+            <mat-icon>edit</mat-icon>
+          </button>
+        </div>
+        <div class="editableInput" *ngIf="showEditableBug">
+          <div>
+            <mat-form-field>
+              <input matInput placeholder="" 
+                [(ngModel)]="bug"
+                (keyup.enter)=
+                "showEditableBug = false; saveBug()"
+                >
+            </mat-form-field>
+          </div>
+          <div>
+            <div
+              class="actions"
+              fxFlex
+              fxLayoutAlign="end end"
+              fxLayoutGap="10px"
+              >
+              <span class="input cancel">
+                <button 
+                  mat-raised-button 
+                  (click)="showEditableBug = false; 
+                  getBug()"
+                  >
+                  Cancel
+                </button>
+              </span>
+              <span class="input save">
+                <button 
+                  mat-raised-button 
+                  color="primary"
+                  (click)="showEditableBug = false; 
+                  saveBug()"
+                  >
+                  Save
+                </button>
+              </span>
+            </div>
+          </div>
+        </div>
+      </ng-container>
+    </div>
+  </div>
```

Comment:
These are huge chunks of code with the same functionality. Could we somehow make it more DRY?

Reply:
Yeah, I'm making these into a separate generic component.